server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java

```diff
@@ -642,9 +668,21 @@ private Optional<AbstractTopologyResponse> getTopologyResponse(short clientIntel
                if (cacheMode.isClustered()) {
                   // Use the request cache's topology id as the HotRod topologyId.
                   int currentTopologyId = cacheTopology.getTopologyId();
-                  // AND if the client's topology id is smaller than the server's topology id
-                  if (topologyId < currentTopologyId)
+                  // AND if the client's topology id is different from the server's topology id
+                  if (topologyId != currentTopologyId) {
+                     ChannelId id = channel.id();
+                     LastKnownClientTopologyId lastKnown = lastKnownTopologyIds.get(id);
+                     if (lastKnown != null && lastKnown.topologyId == topologyId) {
+                        if (timeService.timeDuration(lastKnown.timeCaptured, TimeUnit.SECONDS) > 2) {
```

Comment:
This 2 here is very arbitrary, I can see increasing it higher as well if we think 2 seconds is too low. I personally don't see a reason to make it configurable.

Reply:
+1 to increase.
probably would be nice to have it configurable via `-D` options and a way to disable this?

Reply:
Alright, I can add `infinispan.server.topology.check` and it can be configured in milliseconds and if it is 0 or less it will be disabled.